tools/run_tests/run_interop_tests.py

```diff
@@ -1285,23 +1300,40 @@ def aggregate_http2_results(stdout):
 
     jobs = []
     if args.cloud_to_prod:
-        if args.transport_security != 'tls':
-            print('TLS is always enabled for cloud_to_prod scenarios.')
+        if args.transport_security not in ['tls', 'google_default_credentials']:
+            print(
+                'TLS or google default credential is always enabled for cloud_to_prod scenarios.'
+            )
         for server_host_nickname in args.prod_servers:
             for language in languages:
                 for test_case in _TEST_CASES:
                     if not test_case in language.unimplemented_test_cases():
                         if not test_case in _SKIP_ADVANCED + _SKIP_COMPRESSION:
-                            test_job = cloud_to_prod_jobspec(
+                            tls_test_job = cloud_to_prod_jobspec(
                                 language,
                                 test_case,
                                 server_host_nickname,
                                 prod_servers[server_host_nickname],
                                 docker_image=docker_images.get(str(language)),
                                 manual_cmd_log=client_manual_cmd_log,
                                 service_account_key_file=args.
-                                service_account_key_file)
-                            jobs.append(test_job)
+                                service_account_key_file,
```

Comment:
I think this works. But it now becomes a little odd that we can no longer run the script with <i>only</i> TLS, or <i>only</i> google default creds, they need to now always be ran together.

@yihuazhang @adelez I'm thinking it might end up simpler to modify `grpc_run_interop_tests.sh` to invoke `run_interop_tests.py` twice, with different credentials.

Reply:
`grpc_run_interop_tests.sh` is also invoked by other interop tests including `grpc_interop_alts` and `grpc_interop_tocloud` so we need to modify the code in a way that it also accommodates those two cases. 

Furthermore, for cloud-path test, I think it might be good idea to test TLS and google default credentials together, and the current change in the PR (i.e., `cloud_to_prod` and `cloud_to_prod_auth`) only applies to the cloud-path test.